Resources/Catalog/FileCatalog.py

```diff
@@ -166,8 +184,8 @@ def w_execute( self, *parms, **kws ):
       # Skip if the method is not implemented in this catalog
       if not oCatalog.hasCatalogMethod( self.call ):
         if master:
-          gLogger.error( "Master catalog does not implement the write method", self.call )
-          return S_ERROR( "Master catalog does not implement the write method %s" % self.call )
+          self.log.error( "Master catalog does not implement the write method", self.call )
```

Comment:
I have removed that part in my PR because you can't reach this piece of code. If the method is not in the master, then __getattr__ will throw an exception, so this piece of code can just go